### PR TITLE
Add direct_write option to NpyFileChunkStore

### PR DESCRIPTION
**Pull request by bmerry**
When true, writes the chunks with O_DIRECT option to bypass the OS
filesystem cache.

### CODE REVIEW

**Comment by bmerry** [comment]
This makes the data writer less liable to drop packets.

File: katdal/chunkstore_npy.py [hunk 1]
```diff
@@ -19,10 +19,33 @@
...
 import numpy as np
 
-from .chunkstore import ChunkStore, StoreUnavailable, ChunkNotFound, BadChunk
+from .chunkstore import (ChunkStore, StoreUnavailable, ChunkNotFound, BadChunk,
+                         npy_header_and_body)
+
+
+def _write_chunk(filename, chunk, direct_write):
+    if not direct_write:
+        return np.save(filename, chunk, allow_pickle=False)
```
Comment (ludwigschwardt) [inline katdal/chunkstore_npy.py]: Yay for the happy path! Death to excessive indentation!

File: katdal/test/test_chunkstore_npy.py [hunk 2]
```diff
@@ -42,3 +42,13 @@ def teardown_class(cls):
...
+
+
+class TestNpyFileChunkStoreDirectWrite(TestNpyFileChunkStore):
+    """Test NPY file functionality with O_DIRECT writes."""
+
+    @classmethod
+    def setup_class(cls):
+        """Create temp dir to store NPY files and build ChunkStore on that."""
+        cls.tempdir = tempfile.mkdtemp()
+        cls.store = NpyFileChunkStore(cls.tempdir, direct_write=True)
```
Comment (ludwigschwardt) [inline katdal/test/test_chunkstore_npy.py]: No `os.O_DIRECT` on macOS...

Comment (bmerry) [reply katdal/test/test_chunkstore_npy.py]: Graceful degradation added.

**Comment by ludwigschwardt** [comment]
Great, the test skips on my laptop as expected.

**Pull request merged**